src/cljs/org/broadinstitute/firecloud_ui/page/workspace/create.cljs

```diff
@@ -3,54 +3,57 @@
    [dmohs.react :as react]
    [org.broadinstitute.firecloud-ui.common :as common]
    [org.broadinstitute.firecloud-ui.common.components :as comps]
-   [org.broadinstitute.firecloud-ui.common.dialog :as dialog]
    [org.broadinstitute.firecloud-ui.common.input :as input]
+   [org.broadinstitute.firecloud-ui.common.modal :as modal]
    [org.broadinstitute.firecloud-ui.common.style :as style]
    [org.broadinstitute.firecloud-ui.endpoints :as endpoints]
    [org.broadinstitute.firecloud-ui.nav :as nav]
    [org.broadinstitute.firecloud-ui.utils :as utils]
    ))
 
 
-(react/defc Dialog
+(react/defc CreateDialog
   {:get-initial-state
    (fn [{:keys [props]}]
      {:selected-project (first (:billing-projects props))
       :protected-option :not-loaded})
    :render
-   (fn [{:keys [props state this]}]
-     (dialog/standard-dialog
-       {:width 500
-        :dismiss-self (:dismiss props)
-        :header "Create New Workspace"
-        :ok-button [comps/Button {:text "Create Workspace" :onClick #(react/call :create-workspace this)}]
-        :content
-        [:div {:style {:marginBottom -20}}
-         (when (:creating-wf @state)
-           [comps/Blocker {:banner "Creating Workspace..."}])
-         (style/create-form-label "Billing Project")
-         (style/create-select
-           {:value (:selected-project @state)
-            :onChange #(swap! state assoc :selected-project (-> % .-target .-value))}
-           (:billing-projects props))
-         (style/create-form-label "Name")
-         [input/TextField {:ref "wsName" :style {:width "100%"}
-                           :predicates [(input/nonempty "Workspace name")
-                                        (input/alphanumeric_- "Workspace name")]}]
-         (style/create-textfield-hint "Only letters, numbers, underscores, and dashes allowed")
-         (style/create-form-label "Description (optional)")
-         (style/create-text-area {:style {:width "100%"} :rows 5 :ref "wsDescription"})
-         [:div {:style {:marginBottom "1em"}}
-          [comps/Checkbox
-           {:ref "protected-check"
-            :label "Workspace intended to contain NIH protected data"
-            :disabled? (not= (:protected-option @state) :enabled)
-            :disabled-text (case (:protected-option @state)
-                             :not-loaded "Account status has not finished loading."
-                             :not-available "This option is not available for your account."
-                             nil)}]]
-         [comps/ErrorViewer {:error (:server-error @state)}]
-         (style/create-validation-error-message (:validation-errors @state))]}))
+   (fn [{:keys [props state refs this]}]
+     [modal/OKCancelForm
+      {:header "Create New Workspace"
+       :ok-button (react/create-element
+                    [comps/Button {:ref "okButton" :text "Create Workspace"
+                                   :onClick #(react/call :create-workspace this)}])
```

Comment:
why do you need to create the Button element here, instead of passing a map in to the :ok-button? Is it just to get the ref? If so, is there a way to streamline that, such as defaulting :get-last-element-dom-node to the ok button insdie the OKCancelForm?